### ISSUE TITLE
Crash when trying to get info from EllipseROI in ImageView

### ISSUE THREAD

**Issue by borupdaniel** [pyqtgraph/pyqtgraph#1054]
### Short description
Bit new to Qt so please bear with me if this isn't a complete enough explanation! We have an app that uses Qt and I'm trying to implement some new features using pyqtgraph. One thing I'd love to be able to do is use other ROI shapes in the ImageView widget. I'm trying to do this with an ellipse as an example, but run into a crash the way I've attempted to do this. The crash appears to be related to how `EllipseROI.getArrayRegion` attempts to process the data coming out of `ROI.getArrayRegion`.

### Code to reproduce

```python
import pyqtgraph as pg
import numpy as np

class roiAnalysisBox(gpi.GenericWidgetGroup):
    valueChanged = gpi.Signal(int)
    def __init__(self, title, parent=None):
        super().__init__(title, parent)
        testImage = np.random.random((128,128))
        self.imageView = pg.ImageView()
        self.imageView.setImage(testImage)
        self.imageView.customRoi = pg.EllipseROI(pos = [10,10],size=20)
        self.imageView.view.addItem(self.imageView.customRoi)
        self.imageView.customRoi.show()
        self.imageView.roi = self.imageView.customRoi
        self.imageView.roi.sigRegionChanged.connect(self.imageView.roiChanged)
        wdgLayout = QtWidgets.QHBoxLayout()
        wdgLayout.addWidget(self.imageView)
        self.setLayout(wdgLayout)
```

### Expected behavior

I expect to see an elliptical ROI instead of a rectangular one. I'm able to get this to work if I modify the following lines (1759 and 1760 in ROI.py) to just `return arr`, although I'm not sure what's being returned is correct in that case.
```
        if arr is None or arr.shape[axes[0]] == 0 or arr.shape[axes[1]] == 0:
            return arr
```

### Real behavior

My program crashes and I get the following error message:
```
Traceback (most recent call last):
  File "/Users/m206107/gpi_stack/envs/gpi_coredev/lib/python3.7/site-packages/pyqtgraph/imageview/ImageView.py", line 531, in roiClicked
    self.roiChanged()
  File "/Users/m206107/gpi_stack/envs/gpi_coredev/lib/python3.7/site-packages/pyqtgraph/imageview/ImageView.py", line 567, in roiChanged
    data, coords = self.roi.getArrayRegion(image.view(np.ndarray), self.imageItem, axes, returnMappedCoords=True)
  File "/Users/m206107/gpi_stack/envs/gpi_coredev/lib/python3.7/site-packages/pyqtgraph/graphicsItems/ROI.py", line 1759, in getArrayRegion
    if arr is None or arr.shape[axes[0]] == 0 or arr.shape[axes[1]] == 0:
AttributeError: 'tuple' object has no attribute 'shape'
```

### Tested environment(s)

 * PyQtGraph version: 0.10.0 
 * Qt Python binding: 'PyQt5 5.9.2 Qt 5.9.7'
 * Python version: 3.7.3
 * NumPy version: 1.17.1
 * Operating system: OSX 10.14
 * Installation method: conda

### Additional context
Output of `conda list`
<details>

```
bzip2                     1.0.8                h01d97ff_0    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
certifi                   2019.6.16                py37_1    conda-forge
cycler                    0.10.0                     py_1    conda-forge
dbus                      1.13.6               h2f22bb5_0    conda-forge
eigen                     3.3.7             h04f5b5a_1000    conda-forge
expat                     2.2.5             h6de7cb9_1003    conda-forge
fftw                      3.3.8           nompi_h9629793_1109    conda-forge
freetype                  2.10.0               h24853df_1    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
glib                      2.58.3            h9d45998_1002    conda-forge
gpi                       1.1.5                    py37_0    conda-forge
gpi_core                  2.0.2            py37h6de7cb9_0    conda-forge
h5py                      2.9.0           nompi_py37h6248fd5_1104    conda-forge
hdf5                      1.10.5          nompi_h15a436c_1103    conda-forge
icu                       64.2                 h6de7cb9_1    conda-forge
jpeg                      9c                h1de35cc_1001    conda-forge
kiwisolver                1.1.0            py37h770b8ee_0    conda-forge
libblas                   3.8.0               12_openblas    conda-forge
libcblas                  3.8.0               12_openblas    conda-forge
libcxx                    8.0.1                         0    conda-forge
libcxxabi                 8.0.1                         0    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h01d97ff_1005    conda-forge
liblapack                 3.8.0               12_openblas    conda-forge
libopenblas               0.3.7                h4bb4525_1    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
libtiff                   4.0.10            hd08fb8f_1003    conda-forge
llvm-openmp               8.0.1                h770b8ee_0    conda-forge
llvmlite                  0.29.0           py37h3fea490_1    conda-forge
lz4-c                     1.8.3             h6de7cb9_1001    conda-forge
matplotlib                2.2.4                    py37_0    conda-forge
matplotlib-base           2.2.4            py37h3a684a6_0    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
numba                     0.45.1           py37h86efe34_0    conda-forge
numpy                     1.17.1           py37h6b0580a_0    conda-forge
olefile                   0.46                       py_0    conda-forge
openssl                   1.1.1c               h01d97ff_0    conda-forge
pcre                      8.41              h0a44026_1003    conda-forge
pillow                    6.1.0            py37h75ffe9a_1    conda-forge
pip                       19.2.3                   py37_0    conda-forge
psutil                    5.6.3            py37h01d97ff_0    conda-forge
pyopengl                  3.1.3b2                    py_1    conda-forge
pyparsing                 2.4.2                      py_0    conda-forge
pyqt                      5.9.2            py37h2a560b1_3    conda-forge
pyqtgraph                 0.10.0                     py_5    conda-forge
python                    3.7.3                h93065d6_1    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
pytz                      2019.2                     py_0    conda-forge
qimage2ndarray            1.7                        py_0    conda-forge
qt                        5.9.7                h8cf7e54_3    conda-forge
qtpy                      1.9.0                      py_0    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
scipy                     1.3.1            py37h7e0e109_2    conda-forge
setuptools                41.2.0                   py37_0    conda-forge
sip                       4.19.8          py37h0a44026_1000    conda-forge
six                       1.12.0                py37_1000    conda-forge
sqlite                    3.29.0               hb7d70f7_1    conda-forge
tk                        8.6.9             h2573ce8_1002    conda-forge
tornado                   6.0.3            py37h01d97ff_0    conda-forge
wheel                     0.33.6                   py37_0    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
zlib                      1.2.11            h01d97ff_1005    conda-forge
zstd                      1.4.0                ha9f0a20_0    conda-forge
```

</details>

**Comment by 2xB** [comment]
Could reproduce. Here is an example that can be executed without modification:

```python3
import pyqtgraph as pg
import numpy as np
from PyQt5 import QtWidgets, QtCore
import sys

app = pg.mkQApp()

class roiAnalysisBox(QtWidgets.QGroupBox):
    valueChanged = QtCore.pyqtSignal(int)
    def __init__(self, title, parent=None):
        super().__init__(title, parent)
        testImage = np.random.random((128,128))
        self.imageView = pg.ImageView()
        self.imageView.setImage(testImage)
        self.imageView.customRoi = pg.EllipseROI(pos = [10,10],size=20)
        self.imageView.view.addItem(self.imageView.customRoi)
        self.imageView.customRoi.show()
        self.imageView.roi = self.imageView.customRoi
        self.imageView.roi.sigRegionChanged.connect(self.imageView.roiChanged)
        wdgLayout = QtWidgets.QHBoxLayout()
        wdgLayout.addWidget(self.imageView)
        self.setLayout(wdgLayout)

window = roiAnalysisBox("test")
window.show()
sys.exit(app.exec_())
```

**Comment by 2xB** [comment]
The crash occurs when the ellipse handles are moved.

**Comment by ixjlyons** [comment]
The `ImageView.roiChanged` callback relies on the ROI being able to return the coordinates associated with the selected data (i.e. `returnMappedCoords`), which `EllipseROI` doesn't support currently. One solution for this would be to implement your own `roiChanged` callback in an `ImageView` subclass. You'd have to decide what kind of averaging you want to perform and how to represent that in the `PlotItem`.

Of course we could also implement `returnMappedCoords` for `EllipseROI.getArrayRegion`. I'd have to give that some more thought, but I think it can just return the coordinates from the base `ROI.getArrayRegion`. It essentially returns the data from the bounding rectangle of the ellipse with corners masked out, so the coordinates would just correspond to the rectangle. This would be true of all 2D ROI types if I'm not mistaken.

Also just a note: there was actually a recent PR (#1284) where I decided not to raise an exception if `returnMappedCoords` is requested but not yet implemented. That would've given a clearer error message here, so I'll reconsider that.

**Comment by borupdaniel** [comment]
I'm not actively working on this anymore so just FWIW it's not high priority from my end, but those still sound like some nice ideas for the project!

**Comment by ixjlyons** [comment]
Fixed by #1700
